src/arch/x86/trampoline/mod.rs

```diff
@@ -64,9 +63,28 @@ impl Builder {
   pub unsafe fn build(mut self) -> Result<Trampoline> {
     let mut emitter = pic::CodeEmitter::new();
 
-    while !self.finished {
-      let instruction = self.next_instruction()?;
-      let thunk = self.process_instruction(&instruction)?;
+    // 15 = max size of x64 instruction
+    // safety: we don't know the end address of a function so this could be too far
+    // if the function is right at the end of the code section iced_x86 decoder
+    // doesn't have a way to read one byte at a time without creating a slice in
+    // advance and it's invalid to make a slice that's too long we could make a
+    // new Decoder before reading every individual instruction? but it'd still need
+    // to be given a 15 byte slice to handle any valid x64 instruction
+    let slice = unsafe { &*slice_from_raw_parts(self.target as *const u8, self.margin + 15) };
```

Comment:
@Hpmason this is the bit I'm worried about

Reply:
Yeah, I saw this when looking at the differences between `udis` and `iced-x86`. Switching from disassembly via pointer to a slice will need to be handled differently. I need to look more into `iced-x86`, but didn't have the chance to this weekend. I'll try to look more into it this week to help tackle this.